src/components/sign-up.tsx

```diff
@@ -0,0 +1,228 @@
+import * as React from 'react';
+import { NavigationScreenConfigProps } from 'react-navigation';
+import { KeyboardAvoidingView, View, Text, TextInput, Button, Image, ScrollView } from "react-native";
+
+import Styles from '../styles/main';
+import { config } from '../config';
+import xively from '../lib/xively';
+import { RequestStatus } from '../types/index';
+import * as localAPI from '../lib/local-api/index';
+
+interface SignUpProps extends NavigationScreenConfigProps {
+}
+
+interface SignUpState {
+  requestStatus?: RequestStatus;
+  error?: string;
+  firstName?: string;
+  lastName?: string;
+  email?: string;
+  password?: string;
+  passwordConfirm?: string;
+}
+
+export class SignUpComponent extends React.Component<SignUpProps, SignUpState> {
+  form: any;
+
+  constructor(props: SignUpProps) {
+    super(props);
+
+    this.state = {
+      requestStatus: RequestStatus.REQUEST_NOT_SENT,
+      error: '',
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+    };
+  }
+
+  async submit() {
+    this.setState({ requestStatus : RequestStatus.REQUEST_SENT });
+
+    let userOptions = {
+      emailAddress : this.state.email,
+      password : this.state.password,
+      passwordConfirm : this.state.password,
+      firstName: this.state.firstName,
+      lastName: this.state.lastName,
+    };
+
```

Comment:
here you should at least check that `password` and `confirmPassword` are the same and set error message appropriately.